app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -104,230 +124,137 @@ def available_subnets
-      # unique within a resource group
       vm = vms.all.find { |vm| vm.name == uuid }
       raise ActiveRecord::RecordNotFound unless vm.present?
       vm
     end
 
-    def create_nics(args = {})
-      nics               = []
-      formatted_region = args[:location].gsub(/\s+/, '').downcase
-      args[:interfaces_attributes].each do |nic, attrs|
-        attrs[:pubip_alloc]  = attrs[:bridge]
-        attrs[:privip_alloc] = (attrs[:name] == 'false') ? false : true
-        pip_alloc            = case attrs[:pubip_alloc]
-                                 when 'Static'
-                                   Fog::ARM::Network::Models::IPAllocationMethod::Static
-                                 when 'Dynamic'
-                                   Fog::ARM::Network::Models::IPAllocationMethod::Dynamic
-                                 when 'None'
-                                   nil
-                               end
-        priv_ip_alloc        = if attrs[:priv_ip_alloc]
-                                 Fog::ARM::Network::Models::IPAllocationMethod::Static
-                               else
-                                 Fog::ARM::Network::Models::IPAllocationMethod::Dynamic
-                               end
-        if pip_alloc.present?
-          pip = azure_network_service.public_ips.create(
-              name:                        "#{args[:vm_name]}-pip#{nic}",
-              resource_group:              args[:resource_group],
-              location:                    formatted_region,
-              public_ip_allocation_method: pip_alloc
-          )
-        end
-        new_nic = azure_network_service.network_interfaces.create(
-            name:                         "#{args[:vm_name]}-nic#{nic}",
-            resource_group:               args[:resource_group],
-            location:                     formatted_region,
-            subnet_id:                    attrs[:network],
-            public_ip_address_id:         pip.present? ? pip.id : nil,
-            ip_configuration_name:        'ForemanIPConfiguration',
-            private_ip_allocation_method: priv_ip_alloc
-        )
-        nics << new_nic
-      end
-      nics
+    # user data support
+    def user_data_supported?
+      true
     end
 
-    # Preferred behavior is to utilize Fog but Fog Azure RM
-    # does not currently support creating managed VMs
     def create_vm(args = {})
-      args[:vm_name] = args[:name].split('.')[0]
-      nics           = create_nics(args)
-      if args[:ssh_key_data].present?
+      args[:azure_vm][:vm_name] = args[:name].split('.')[0]
+      nics = create_nics(args)
+      if args[:azure_vm][:password].present? && !args[:azure_vm][:ssh_key_data].present?
+        args[:azure_vm][:script_command] = "$echo #{args[:azure_vm][:password]} | sudo -S echo '\"#{args[:azure_vm][:username]}\" ALL=(ALL) NOPASSWD: ALL' > /etc/sudoers.d/waagent"
```

Comment:
Here, we are setting the value of `args[:azure_vm][:script_command]`. But, how should we handle this when even the user will pass some data in this field, would it make better sense if we concatenate both strings internally?

Reply:
Yes, we should concatenate the strings. We need both user script and adding the user to sudoers.

Reply:
+1
Pushed the change.